src/WarcraftLegacies.Source/Quests/QuestTombOfSargeras.cs

```diff
@@ -26,18 +30,15 @@ public sealed class QuestTombOfSargeras : QuestData
       _tombOfSargerasInteriorEntrance = tombOfSargerasInteriorEntrance;
       _guldanRemains = guldanRemains;
       AddObjective(new ObjectiveTime(900));
-      AddObjective(new ObjectiveEitherOf(
-        new ObjectiveLegendReachRect(LegendLordaeron.Uther, Regions.Sargeras_Entrance,
-          "the Tomb of Sargeras' entrance"),
-        new ObjectiveHeroWithLevelReachRect(10, Regions.Sargeras_Entrance, "the Tomb of Sargeras' entrance")));
+      AddObjective(new ObjectiveHeroWithLevelReachRect(10, Regions.Sargeras_Entrance, "the Tomb of Sargeras' entrance"));
+      Global = true;
     }
     
     /// <inheritdoc />
     protected override string RewardDescription => "The Tomb of Sargeras opens";
 
-    //Todo; it would be really cool if this depended on the hero that opens it
     /// <inheritdoc />
-    protected override string CompletionPopup => "The Tomb of Sargeras has been opened.";
+    protected override string CompletionPopup => $"The Tomb of Sargeras has been opened by {GetTriggerUnit().GetProperName()}";
```

Comment:
This is brilliant, but is also a slightly risky way to do it. We cannot guarantee that the last Objective completed is ObjectiveHeroWithLevelReachRect, so we might not have a valid TriggerUnit(). 
Instead I would suggest modifying ObjectiveHeroWithLevelReachRect to set a public property on itself whenever it gets completed, then use that public property here.

Reply:
I hope I solved that the way you wanted to.
I added `public string EnteredRegionUnitName { get; private set; } = "an unknown hero.";` inside the `ObjectiveHeroWithLevelReachRect` and then assigned that in the `private void OnRegionEnter()` 
I then added a private field to `QuestTombOfSargeras` which stores the objective so I can use the name in the `CompletionPopup`